src/VisualStudio/Core/Def/Implementation/TableDataSource/Suppression/VisualStudioSuppressionFixService.cs

```diff
@@ -110,7 +143,7 @@ public bool RemoveSuppressions(bool selectedErrorListEntriesOnly)
                 }
 
                 var project = solution.GetProject(diagnosticsByProject.Key);
-                if (project != null)
+                if (project != null && shouldFixInProject(project))
```

Comment:
and, on the plus side, you don't need the null check here, it can just be in ShouldFixInProject